### ISSUE TITLE
0.10.4: shpinx fails

### ISSUE THREAD

**Issue by kloczek** [beartype/beartype#120]
Looks like lates sphinx fails on generate documentation
```console
+ /usr/bin/python3 setup.py build_sphinx -b man --build-dir build/sphinx
running build_sphinx
Running Sphinx v4.5.0

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/sphinx/config.py", line 332, in eval_config_file
    exec(code, namespace)
  File "/home/tkloczko/rpmbuild/BUILD/beartype-0.10.4/doc/source/conf.py", line 45, in <module>
    from beartype.roar import BeartypeDependencyOptionalMissingWarning
ImportError: cannot import name 'BeartypeDependencyOptionalMissingWarning' from 'beartype.roar' (/home/tkloczko/rpmbuild/BUILD/beartype-0.10.4/beartype/roar/__init__.py)
```

**Comment by leycec** [comment]
**Hah!** The joke's on you, @[kloczek](https://github.com/kloczek)! We [don't even *have* Sphinx documentation](https://github.com/beartype/beartype/issues/120). It can't be broken if it doesn't exist. Checkmate. :chess_pawn: 

In all seriousness, Sphinx is a unique Hell we're currently working on fixing. Sphinx-based documentation is in-process and well-known to be broken. Yes, this is shameful. Yes, this reflects poorly on @beartype. That's why issue #8 is our highest priority. For the moment, please pretend that we have no `doc/` directory. I can only insert mournful emoji here. :disappointed: :sob: 

Oh – and thanks for letting us know, nonetheless. If we thought we actually had working documentation, I'd be ruthlessly hell-bent on fixing this! Since you're awesome, @kloczek, just let us know if anything else looks brokey.

**Comment by kloczek** [comment]
Just testes 0.11.0 and documentation builds with small parch
```patch
--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -41,8 +41,9 @@
 # See also:
 # * https://samnicholls.net/2016/06/15/how-to-sphinx-readthedocs
 #   The "Make autodoc actually work" is the canonical writeup on this kludge.
-# import os, sys
-# # sys.path.insert(0, os.path.abspath('../../'))
+import os
+import sys
+sys.path.insert(0, os.path.abspath("../../"))
 # print(f'sys.path (from "doc/source/conf.py"): {sys.path}')

 # ....................{ IMPORTS                           }....................
```
With that
```console
+ /usr/bin/sphinx-build -n -T -b man doc/source build/sphinx/man
Running Sphinx v5.1.1
WARNING: Optional Sphinx extension "sphinx_rtd_theme" not found; falling back to default Sphinx HTML theme.
making output directory... done
WARNING: html_static_path entry '_static' does not exist
[autosummary] generating autosummary for: changes.rst, index.rst, reference.rst, tutorial.rst
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: [new config] 4 added, 0 changed, 0 removed
reading sources... [100%] tutorial
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... beartype.1 { changes reference tutorial } /home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/__init__.py:docstring of beartype:3: WARNING: py:mod reference target not found: beartype.meta
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:class reference target not found: beartype._data.datatyping.BeartypeableT
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:class reference target not found: beartype._data.datatyping.BeartypeableT
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:class reference target not found: beartype._data.datatyping.BeartypeableT
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:class reference target not found: BeartypeableT
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:class reference target not found: BeartypeConf
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:class reference target not found: optional
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:class reference target not found: BeartypeReturn
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:exc reference target not found: BeartypeConfException
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:67: WARNING: py:class reference target not found: BeartypeConf
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:exc reference target not found: BeartypeDecorHintException
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:68: WARNING: py:mod reference target not found: typing
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:exc reference target not found: BeartypePep563Exception
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:exc reference target not found: BeartypeDecorParamNameException
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:exc reference target not found: BeartypeDecorWrappeeException
/home/tkloczko/rpmbuild/BUILD/beartype-0.11.0/beartype/_decor/_cache/cachedecor.py:docstring of beartype._decor._cache.cachedecor.beartype:1: WARNING: py:exc reference target not found: BeartypeDecorWrapperException
done
build succeeded, 17 warnings.
```
You can peak on fixes that kind of issues in other projects
https://github.com/latchset/jwcrypto/pull/289
https://github.com/click-contrib/sphinx-click/commit/abc31069
https://github.com/latchset/jwcrypto/pull/289
https://github.com/RDFLib/rdflib-sqlalchemy/issues/95
https://github.com/sissaschool/elementpath/commit/bf869d9e
https://github.com/jaraco/cssutils/issues/21
https://github.com/pywbem/pywbem/pull/2895
https://github.com/sissaschool/xmlschema/commit/42ea98f2
https://github.com/RDFLib/rdflib/pull/2036
https://github.com/frostming/unearth/issues/14
https://github.com/pypa/distlib/commit/98b9b89f

**Comment by leycec** [comment]
Oh, nice! Thanks a fistful of fallen Autumn leaves for both that fast patch *and* the generous helping of links from fellow Sphinx ~~victims~~ survivors. I must confess I haven't touched Sphinx with a ten-foot septic pole after becoming irrationally upset and pounding angrily on my keyboard five months ago.

What I'm trying to say is... **re-resolved by commit 5598744.** :hot_face:

**Comment by kloczek** [comment]
[mode=bowing]
Thank you 👍 😄 
[/mode]
